contracts/PositionToken.sol

```diff
@@ -3,20 +3,64 @@ pragma solidity 0.8.13;
 
 import "hardhat/console.sol";
 
-import {ERC721} from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
-import {ERC721Burnable} from "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import "@openzeppelin/contracts/token/ERC721/extensions/ERC721Burnable.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
 
-contract PositionToken is ERC721, ERC721Burnable {
-    constructor() ERC721("PositionToken", "PNT") {}
+contract PositionToken is ERC721, ERC721Burnable, AccessControl {
+    using Counters for Counters.Counter;
 
-    /* Privileged functions: Executive  */
+    Counters.Counter private _tokenIdCounter;
 
-    /// @dev Mints NFT tokens to a recipient.
-    ///
-    /// This function reverts if the caller does not have the minter role.
-    ///
-    /// @param _recipient the account to mint tokens to.
-    function mint(address _recipient, uint256 tokenId) external {
-        _mint(_recipient, tokenId);
+    address private _leverageEngineAddress;
+
+    bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");
+    bytes32 public constant EXECUTIVE_ROLE = keccak256("EXECUTIVE_ROLE");
+
+    bool internal _initialized = false;
+
+    modifier onlyAdmin() {
+        require(hasRole(ADMIN_ROLE, msg.sender), "onlyAdmin: Caller is not admin");
+        _;
+    }
+
+    modifier onlyExecutive() {
+        require(hasRole(EXECUTIVE_ROLE, msg.sender), "onlyExecutive: Caller is not executive");
+        _;
+    }
+
+    modifier expectInitialized() {
+        require(_initialized, "expectInitialized: contract is not initialized");
+        _;
+    }
+
+    constructor(address admin) ERC721("PositionToken", "PNT") {
+        _setupRole(ADMIN_ROLE, admin);
+    }
+
+    function init(address leverageEngine) public onlyAdmin {
+        _setupRole(EXECUTIVE_ROLE, leverageEngine);
+        _leverageEngineAddress = leverageEngine;
```

Comment:
executive role will be leverageEngine since it should be the only contract that needs to execute methods other than the owner.